### PR TITLE
[WIP] [kde-plasma/kscreenlocker] Add live ebuild

### PR DESCRIPTION
**Pull request by eliasp**
This is a preliminary ebuild for `kde-plasma/kscreenlocker-9999`.

**TODO:**
- [x] Find a fix for the missing `$ENV{DESTDIR}` in `kcheckpass/CMakeLists.txt` - causes `chown` and `chmod` of `kcheckpass` to fail
- [x] Add dependencies in `kde-plasma/workspace-9999` and `kde-plasma/kwin-9999`
- [x] Add blocker for `<kde-plasma/plasma-workspace-5.4.50`
- [x] Add autotests
- [x] Check setuid requirements for `kcheckpass` with `USE="-pam"`
- [ ] Add blocker on `kde-base/kcheckpass` in:
  - [ ] `kde-plasma/kscreenlocker`
  - [ ] `kde-plasma/plasma-workspace`
- [ ] Check for further transitive deps using `lddtree` and add them
- [x] Check `CMakeLists.txt` for further Qt deps

### CODE REVIEW

**Comment by a17r** [comment]
Ebuild will also need to block conflicting versions of plasma-workspace, <5.4.50 should suffice.

File: kde-plasma/kscreenlocker/kscreenlocker-9999.ebuild [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	$(add_frameworks_dep kidletime)
+	$(add_frameworks_dep plasma)
+	$(add_plasma_dep kwayland)
+	dev-libs/wayland
+	dev-qt/qtx11extras:5
+	x11-libs/libX11
+	x11-libs/libXi
+	x11-libs/libxcb
+	x11-libs/xcb-util-keysyms
+	!<kde-plasma/plasma-workspace-5.4.50
```
Comment (a17r) [inline kde-plasma/kscreenlocker/kscreenlocker-9999.ebuild]: The block should not end up in DEPEND, so better to put all the rest into COMMON_DEPEND.

Comment (eliasp) [inline kde-plasma/kscreenlocker/kscreenlocker-9999.ebuild]: Fixed

File: kde-plasma/kscreenlocker/kscreenlocker-9999.ebuild [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	$(add_frameworks_dep kcmutils)
+	$(add_frameworks_dep kcrash)
+	$(add_frameworks_dep kdeclarative)
+	$(add_frameworks_dep kdelibs4support)
+	$(add_frameworks_dep kglobalaccel)
+	$(add_frameworks_dep kidletime)
+	$(add_frameworks_dep plasma)
+	$(add_plasma_dep kwayland)
+	dev-libs/wayland
+	dev-qt/qtx11extras:5
```
Comment (a17r) [inline kde-plasma/kscreenlocker/kscreenlocker-9999.ebuild]: I'm sure there are more Qt deps when looking at CMakeLists.txt, in the case of qtwidgets I'm not sure if it isn't a runtime dependency only.

File: kde-plasma/kscreenlocker/kscreenlocker-9999.ebuild [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+IUSE=""
+
+COMMON_DEPEND="
+	$(add_frameworks_dep kcmutils)
+	$(add_frameworks_dep kcrash)
+	$(add_frameworks_dep kdeclarative)
+	$(add_frameworks_dep kdelibs4support)
+	$(add_frameworks_dep kglobalaccel)
+	$(add_frameworks_dep kidletime)
+	$(add_frameworks_dep plasma)
```
Comment (a17r) [inline kde-plasma/kscreenlocker/kscreenlocker-9999.ebuild]: lddtree should reveal additional transitive deps, when I looked into it I had found at least kwindowsystem, kxmlgui, ki18n and knotifications.

**Comment by a17r** [comment]
Package has autotests that should be made optional, KDE_TEST in kde5.eclass has recently gained "forceoptional" option that should do it all automatically for you.

**Comment by eliasp** [comment]
ReviewRequest for fixing the `$ENV{DESTDIR}` issue: https://git.reviewboard.kde.org/r/126037/

**Comment by a17r** [comment]
Taken from kcheckpass readme:
Compile with -DHAVE_VSYSLOG if you have vsyslog().
Compile with -DHAVE_PAM if you have a PAM system, and link with -lpam -ldl (If libdl is present).
Compile with -DHAVE_SHADOW if you have a shadow password system.

**Comment by a17r** [comment]
Based on the findings in
https://bugs.gentoo.org/show_bug.cgi?id=561660
https://forums.gentoo.org/viewtopic-p-7771962.html#7771962
it might be a good thing to also block kde-base/kcheckpass.

File: kde-plasma/kscreenlocker/kscreenlocker-9999.ebuild (hunk 2 above)
Comment (kensington) [inline kde-plasma/kscreenlocker/kscreenlocker-9999.ebuild]: kscreenlocker_greet additionally links to Qt5Declarative, Qt5Gui, andQt5Network. screenlocker_kcm.so additionally links to Qt5DBus and Qt5Widgets.

File: kde-plasma/kscreenlocker/kscreenlocker-9999.ebuild (hunk 3 above)
Comment (kensington) [inline kde-plasma/kscreenlocker/kscreenlocker-9999.ebuild]: depcheck agrees, suggesting these additional links: KF5Config, KF5ConfigWidgets,KF5CoreAddons, KF5I18n, KF5Notifications, KF5Package, KF5XmlGui, and KF5WindowsSystem.

**Comment by kensington** [comment]
I can confirm the PAM linkage on my system.

**Comment by kensington** [comment]
@a17r there's ```greeter/autotests``` too which won't be handled by the eclass. I wonder if we should make it recursive, introduce some KDE_TEST_DIRS=() variable, or just handle it manually in individual packages (but that's for another PR).

**Comment by eliasp** [comment]
As my available time is quite limited at the moment, feel free to pick up this PR and continue working on it… I might (or not) continue Friday evening.

**Comment by a17r** [comment]
See also updates in https://github.com/gentoo/kde/pull/475 to avoid file collisions (removed pam files from plasma-workspace-9999).

**Comment by kensington** [comment]
Thanks, squashed as ba383e5f6a02e57c395d8f349521c28a3b6de356.

**Pull request closed without merging**